plugin/trino-pulsar/src/main/java/io/trino/plugin/pulsar/decoder/json/PulsarJsonFieldDecoder.java

```
@@ -0,0 +1,438 @@
+        }
+
+        @Override
+        public final boolean isNull()
+        {
+            return value.isMissingNode() || value.isNull();
+        }
+
+        @Override
+        public boolean getBoolean()
+        {
+            return getBoolean(value, columnHandle.getType(), columnHandle.getName());
+        }
+
+        @Override
+        public double getDouble()
+        {
+            return getDouble(value, columnHandle.getType(), columnHandle.getName());
+        }
+
+        @Override
+        public Slice getSlice()
+        {
+            return getSlice(value, columnHandle.getType());
+        }
+
+        public static boolean getBoolean(JsonNode value, Type type, String columnName)
+        {
+            if (value.isValueNode()) {
+                return value.asBoolean();
+            }
+            throw new TrinoException(DECODER_CONVERSION_NOT_SUPPORTED, format("could not parse non-value node as '%s' for column '%s'", type, columnName));
+        }
+
+        public static long getLong(JsonNode value, Type type, String columnName, long minValue, long maxValue)
+        {
+            try {
+                if (type instanceof RealType) {
+                    return floatToIntBits((Float) parseFloat(value.asText()));
+                }
+
+                long longValue;
+                if (value.isIntegralNumber() && !value.isBigInteger()) {
+                    longValue = value.longValue();
+                    if (longValue >= minValue && longValue <= maxValue) {
+                        return longValue;
+                    }
+                }
+                else if (value.isValueNode()) {
+                    longValue = parseLong(value.asText());
+                    if (longValue >= minValue && longValue <= maxValue) {
+                        return longValue;
+                    }
+                }
+            }
+            catch (NumberFormatException ignore) {
+                // ignore
```

please do not swallow exceptions